massiv-examples/GameOfLife/app/GameOfLife.hs

```diff
@@ -30,7 +30,9 @@ life = compute . A.mapStencil Wrap lifeStencil
 initLife :: Sz2 -> Array S Ix2 Word8 -> Array S Ix2 Word8
 initLife sz arr =
   compute $
-  insertWindow (A.replicate Par sz 0) (Window ix0 (size arr) (index' arr . subtract ix0) Nothing)
+  insertWindow
+    (delay $ computeAs A.S $ A.replicate Par sz 0)
```

Comment:
I also found `A.makeArray Par sz (\_ -> 0)` to be a valid parameter for insertWindow. Wouldn't that be more efficient?

Reply:
Normally it would not be more efficient, because `replicate` is implemented with a much more efficient `setByteArray` and when the argument is `0` it is even more efficient, but in this case I think you are right and it will be because it will avoid allocating the whole array unnecessarily.

I think I need to add `replicate` to `Construct` class this way it can be used in the most efficient matter depending on the representation.

Reply:
Swapped for `makeArray`. Thanks